src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs

```diff
@@ -63,64 +60,64 @@ protected Regex()
 
         internal Regex(string pattern, CultureInfo? culture)
         {
-            // Call Init directly rather than delegating to a Regex ctor that takes
-            // options to enable linking / tree shaking to remove the Regex compiler
-            // and NonBacktracking implementation if it's not used.
-            Init(pattern, RegexOptions.None, s_defaultMatchTimeout, culture ?? CultureInfo.CurrentCulture);
-        }
+            // Validate and store the arguments.
+            ValidatePattern(pattern);
```

Comment:
NIT: Any reason why these calls which still get executed by the overload can't go into a centralized Init call to avoid the duplication? Then the other overload would just handle and validate options but all of the shared code would still be centralized.

Reply:
Why calls specifically?

Reply:
seems like both of the internal ctors are doing:

```c#
            ValidatePattern(pattern);
            this.pattern = pattern;
            culture ??= CultureInfo.CurrentCulture; // This should probably be a param to the Init
            internalMatchTimeout = s_defaultMatchTimeout; // This should be a param of the init

            // Parse the pattern.
            RegexTree tree = RegexParser.Parse(pattern, roptions, culture);

            // Store the relevant information, including a factory for using the interpreter.
            capnames = tree.CapNames;
            capslist = tree.CapsList;
            caps = tree.Caps;
            capsize = tree.CapSize;
```

Reply:
Ok